### ISSUE TITLE
assertion_on_constants: False positive on debug_assert!(false, ...)

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#3948]
The following code currently triggers assertion_on_constants. I believe the specific pattern of `debug_assert!(false, ...)` should be ignored by this lint as there is no current better similarly concise way to express the behavior of panic-in-debug-mode-only.

```rust
fn main() {
    debug_assert!(false, "oh no");
}
```

Example occurrence in Chrome OS: [*platform/crosvm/gpu_display/src/lib.rs*](https://chromium.googlesource.com/chromiumos/platform/crosvm/+/d49adc9005a300dbae60bd8ecb12ea620fc0fd31/gpu_display/src/lib.rs#308)

Mentioning @Arkweid and @flip1995 who have worked on this lint.

---

<kbd>clippy 0.0.212 (37f5c1e 2019-04-09)</kbd>
